### PR TITLE
fix(railplus): add PDF overlays for it and fr, update default export size and format for railplus

### PR DESCRIPTION
**Pull request by danji90**
Add PDF overlays for it and fr, update default export size and format for railplus

# How to

# Others

- [ ] It's not a hack or at least an unauthorized hack :).
- [ ] The images added are optimized.
- [ ] Everything in ticket description has been fixed.
- [ ] The author of the MR has made its own review before assigning the reviewer.
- [ ] The title means something for a human being and follows the [conventional commits](https://www.conventionalcommits.org/) specification.
- [ ] The title contains [WIP] if it's necessary.
- [ ] Labels applied. if it's a release? a hotfix?
- [ ] Tests added.

### CODE REVIEW

File: src/components/MapControls/MapControls.js [hunk 1]
```diff
@@ -47,18 +47,24 @@ const MapControls = ({
...
+  const [, setForceRender] = useState(0);
   const setGeolocFeatureWithRef = (feature) => {
     featureRef.current = feature;
     setGeolocationFeature(feature);
   };
 
+  useEffect(() => {
+    // Ensure the translations are always applied
+    setForceRender((prevcount) => prevcount + 1);
+  }, [i18n.language]);
```
Comment (oterral) [inline src/components/MapControls/MapControls.js:66]: If you have to do this there is probably a problem somewhere else . What is the problem you are trying to fix ?

File: src/components/ExportButton/exportUtils.js [hunk 2]
```diff
@@ -324,9 +329,14 @@ export const exportPdf = async (
   // Add canvas to PDF
   doc.addImage(canvas, 'JPEG', 0, 0, exportSize[0], exportSize[1]);
 
+  const topicFileName = topic?.exportConfig?.exportFileName;
+
   // download the result
   const filename = `${
-    topic?.exportConfig?.exportFileName ||
```
Comment (oterral) [inline src/components/ExportButton/exportUtils.js:329]: The exportConfig object makes me nervous in this function :D   can you add exportFileName and overlyImageUrl as parameters) to this function . So we remove the topic  and the i18n object as parameter. then the translation happens at the component level where we already have i18n and t

File: src/config/topics.js [hunk 3]
```diff
@@ -437,10 +439,20 @@ export const railPlus = {
   exportConfig: {
     publishedAt: '11/2023',
     year: '2023',
-    overlayImageUrl: railplusLegend,
-    exportFileName: `RAILplus Streckennetz ${new Date()
-      .toISOString()
-      .slice(0, 10)}`,
+    overlayImageUrl: (i18n) => {
```
Comment (oterral) [inline src/config/topics.js:442]: just pass the language we don't need all the object.

**Review by oterral — CHANGES_REQUESTED** [review]
lgtm minor changes to simplify a bit

File: src/components/MapControls/MapControls.js (hunk 1 above)
Comment (danji90) [reply src/components/MapControls/MapControls.js:66]: When loading railplus topic the mapcontrols fail to rerender after the initial render, even if the language changes in the app state

Comment (danji90) [reply src/components/MapControls/MapControls.js]: fixed on master

File: src/components/ExportButton/exportUtils.js (hunk 2 above)
Comment (danji90) [reply src/components/ExportButton/exportUtils.js:329]: Done

File: src/config/topics.js (hunk 3 above)
Comment (danji90) [reply src/config/topics.js]: ok

**Comment by danji90** [comment]
@oterral Please check the last two commits

File: src/components/ExportButton/ExportButton.js [hunk 4]
```diff
@@ -80,15 +80,39 @@ function ExportButton({
...
+
+        const { exportConfig } = topic;
+        let imageUrl;
+        let fileName = `trafimage-${new Date().toISOString().slice(0, 10)}.pdf`;
+
+        if (exportConfig) {
+          const { overlayImageUrl, exportFileName } = exportConfig;
+          imageUrl =
+            overlayImageUrl &&
+            (typeof overlayImageUrl === 'function'
```
Comment (oterral) [inline src/components/ExportButton/ExportButton.js:92]: make it always a function, it's simplifies

File: src/components/ExportButton/ExportButton.js [hunk 5]
```diff
@@ -80,15 +80,39 @@ function ExportButton({
...
+              ? overlayImageUrl(i18n.language)
+              : overlayImageUrl);
+
+          fileName = exportFileName
```
Comment (oterral) [inline src/components/ExportButton/ExportButton.js:96]: make it always a function getOverlayImgUrl, it's simplifies

File: src/components/ExportButton/exportUtils.js [hunk 6]
```diff
@@ -243,16 +245,16 @@ export const exportPdf = async (
   const ctx = canvas.getContext('2d');
 
   // Apply SVG overlay if provided
-  if (topic.exportConfig && topic.exportConfig.overlayImageUrl) {
-    const { overlayImageUrl, dateDe, dateFr, publisher, publishedAt, year } =
-      topic.exportConfig;
+  if (exportConfig && overlayImageUrl) {
+    const { dateDe, dateFr, publisher, publishedAt, year } = exportConfig;
```
Comment (oterral) [inline src/components/ExportButton/exportUtils.js:249]: not generic enough if we have a new topic , we don't want to always add new values to replace.

in exportConfig create  a  getTemplateValues function that returns an  object : 
getTemplateValues() {
   return {
       year: '2023',
       publishedAt: '2023,    
       dateFr: '...'
       dateDe: '...
}

and then in exportUtils you just go throught these values are replace them in the svg
}

**Review by oterral — CHANGES_REQUESTED** [review]
Make it more generic

File: src/components/ExportButton/ExportButton.js [hunk 7]
```diff
@@ -80,15 +80,39 @@ function ExportButton({
...
+        await exportPdf(
           mapToExport,
           map,
           layers,
           exportFormat,
           canvas,
-          topic,
           exportScale,
           exportSize,
+          exportConfig,
```
Comment (oterral) [inline src/components/ExportButton/ExportButton.js:113]: use a templateValues parameter

Comment (danji90) [reply src/components/ExportButton/ExportButton.js]: done

File: src/components/ExportButton/exportUtils.js (hunk 6 above)
Comment (danji90) [reply src/components/ExportButton/exportUtils.js]: done

File: src/components/ExportButton/ExportButton.js (hunk 5 above)
Comment (danji90) [reply src/components/ExportButton/ExportButton.js]: done

**Review by oterral — APPROVED** [review]
lgtm

**Pull request merged**